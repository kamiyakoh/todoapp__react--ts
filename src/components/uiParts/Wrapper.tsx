import type { ReactNode } from 'react';
import type { SerializedStyles } from '@emotion/react';
/** @jsxRuntime classic */
/** @jsx jsx */
import { jsx, css } from '@emotion/react';

interface Props {
  cssName?: SerializedStyles;
  children: ReactNode;
}

export const Wrapper = ({ cssName, children }: Props): JSX.Element => {
  return <div css={[wrapper, cssName]}>{children}</div>;
};

const wrapper = css`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  gap: 2em;
  padding: 2em 0;
`;
