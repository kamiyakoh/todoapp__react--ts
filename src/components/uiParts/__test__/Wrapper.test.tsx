import type { SerializedStyles } from '@emotion/react';
/** @jsxRuntime classic */
/** @jsx jsx */
import { css, jsx } from '@emotion/react';
import { render } from '@testing-library/react';
import { Wrapper } from '../Wrapper';

const margin = css`
  margin: 80px;
`;

const minHeight = css`
  min-height: 100vh;
`;

describe('Wrapper Test', () => {
  test.each`
    cssName
    ${undefined}
    ${[margin, minHeight]}
  `('render', (cssName: SerializedStyles | SerializedStyles[]) => {
    const { asFragment } = render(<Wrapper cssName={cssName}>children</Wrapper>);
    expect(asFragment()).toMatchSnapshot();
  });
});
