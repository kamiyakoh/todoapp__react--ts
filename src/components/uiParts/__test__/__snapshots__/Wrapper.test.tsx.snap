// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Test render 1`] = `
<DocumentFragment>
  <div
    class="css-5h8oie-Wrapper"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Wrapper Test render 2`] = `
<DocumentFragment>
  <div
    class="css-1nmx4lh-Wrapper"
  >
    children
  </div>
</DocumentFragment>
`;
